src/test/java/com/google/firebase/testing/TestUtils.java

```diff
@@ -85,6 +86,30 @@ public static void setEnvironmentVariables(Map<String, String> vars) {
     }
   }
 
+  public static void unsetEnvironmentVariables(Set<String> vars) {
```

Comment:
This is the same way that we set env variables in tests. We should use this in a lot of other places too, else environment variables can get carried over between test methods.
Currently I use this only for my new tests.

Reply:
This really sucks. Would you consider modifying the code to take env vars as a map in constructors, methods, etc. instead?

Reply:
Done!

Reply:
Not done? I still see we're messing with env vars in tests.

Reply:
Yes we're still doing it in some tests which are sort of "integration" tests for FirebaseDatabase: here we must set the environment variable to test effectively :(